apps/pkey.c

```diff
@@ -258,8 +258,8 @@ int pkey_main(int argc, char **argv)
                            ERR_reason_error_string(err));
                 ERR_get_error(); /* remove err from error stack */
             }
+            goto end;
         }
-        EVP_PKEY_CTX_free(ctx);
```

Comment:
You could have moved this line to appear before `if (r == 1) {` (line 245).

Reply:
It's not the best for debugging, as if one wants to inspect the ctx in case of error it has already been freed.

I can of course comply with your preference if you think it would make things more readable (I imagine this would also mean putting back the declaration of ctx inside this block rather than outside).

Reply:
Er, hmm?  Oh, yeah, ok, I can understand debugability.  That's actually sufficient explanation for the modification you made, thank you.